Source/SourceKittenFramework/Clang+SourceKitten.swift

```diff
@@ -214,20 +214,36 @@ extension CXCursor {
         guard let usr = usr(),
             let findUSR = try? Request.findUSR(file: swiftUUID, usr: usr).send(),
             let usrOffset = findUSR[SwiftDocKey.offset.rawValue] as? Int64 else {
-            return nil
+                return nil
         }
 
         guard let cursorInfo = try? Request.cursorInfo(file: swiftUUID, offset: usrOffset, arguments: compilerArguments).send() else {
             return nil
         }
 
+        return cursorInfo
+    }
+
+    func swiftDeclaration(compilerArguments: [String]) -> String? {
+        guard let cursorInfo = getCursorInfo(compilerArguments: compilerArguments) else {
```

Comment:
Making the SourceKit `cursorInfo` request is expensive, and this PR doubles the number of those requests sent.

Please refactor so that you only need to make a single request per declaration.

Reply:
Yeah was thinking about that. Done now.